### ISSUE TITLE
[Android] Inconsistent first TalkBack focus on navigating to new page

### ISSUE THREAD

**Issue by rachelkang** [dotnet/maui#14187]
### Description

When running my app with TalkBack activated, I notice that the first item to receive focus upon navigating to a new page will vary. Sometimes, it'll navigate to the back button; sometimes, to the first heading; sometimes to the first item on the page. This inconsistency is confusing and a pattern should be standardized.

### Steps to Reproduce

1. Download https://github.com/rachelkang/recipeSearch
2. Deploy to physical Android device
3. Turn on TalkBack
4. Investigate 3 different scenarios in which we'd expect the same behavior:

### Scenario 1
5. Click on a recipe search button like "Breakfast"
6. Click on the first recipe
7. Notice that when navigating to the recipe page, screen reader focus starts on the back button

### Scenario 2
8. Go back
9. Click on any other recipe
10. Notice that when navigating to the recipe page, screen reader focus starts on the "Ingredients:" heading label

### Scenario 3
11.  Click on "My Recipes" tab
12. Click on the first recipe
13. Notice that when navigating to the recipe page, screen reader focus starts on the recipe title "달걀말이 (Korean Egg Roll)"

We would expect focus to start at the same control in all these scenarios, especially as the pages all have the exact same layout

### Link to public reproduction project repository

https://github.com/rachelkang/recipeSearch

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

**Comment by jinxinjuan** [comment]
Verified this issue with Visual Studio Enterprise 17.7.0 Preview 1.0. Not repro on Android Device with sample project.
https://github.com/rachelkang/recipeSearch.
